packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -299,10 +299,15 @@ const PipelineWrapper: React.FC<IProps> = ({
     [runtimeImages]
   );
 
+  const isDialogAlreadyShowing = useRef(false);
   const onError = useCallback(
-    (error?: Error): void => {
+    async (error?: Error): Promise<void> => {
+      if (isDialogAlreadyShowing.current) {
+        return; // bail, we are already showing a dialog.
+      }
+      isDialogAlreadyShowing.current = true;
       if (error instanceof PipelineOutOfDateError) {
-        showDialog({
+        const result = await showDialog({
```

Comment:
Is there a reason you're removing the Promise then handling and replacing it with await result handling? It seems unnecessary for this boolean flag solution to this problem.